### PR TITLE
AVRO-3003: Fully qualify enum default value in C# code gen

### PR DESCRIPTION
**Pull request by massada**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO-3003)
  - https://issues.apache.org/jira/browse/AVRO-3003
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
- Add a new test case to the Avro.Test.AvroGen.GenerateSchemaCheckFields unit test

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Comment by zcsizmadia** [comment]
Please take a look at #1595. It adds many unit tests tfor avrogen and codegen, including code compilation. Very similar fashion you do in your PR. I recommend that after #1595 PR is merged, you use that test infrastructure.

**Comment by zcsizmadia** [comment]
Thanks for picking up that ticket and fixing it!

**Comment by massada** [comment]
#1595 looks great, I'll update this PR once that one is merged

**Comment by zcsizmadia** [comment]
I noticed that you package reference'd Microsoft.CodeAnalysis.CSharp v2.10.0 vs v4.1.0 (latest). Is there any reason for that? Are you aware of some issues with the newer versions?

**Comment by massada** [comment]
Not that I know of, that was the version that Rider added by default, I didn't notice that it selected an older version. I believe that the latest should work just fine.

**Comment by KyleSchoonover** [comment]
@massada Can you do the minor updates?  This looks good to me.

**Comment by massada** [comment]
@KyleSchoonover I'm waiting for #1595 to rebase since there's duplicated work.

File: lang/csharp/src/apache/test/Avro.test.csproj [hunk 1]
```diff
@@ -32,6 +32,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="2.10.0" />
```
[the remark this answers is not in the document]
Comment (massada) [reply lang/csharp/src/apache/test/Avro.test.csproj]: I've rebased and this change is gone

File: lang/csharp/src/apache/test/CodGen/CodeGenTest.cs [hunk 2]
```diff
@@ -19,9 +19,12 @@
 using System.Collections.Generic;
 using System.IO;
 using System.CodeDom.Compiler;
+using System.Reflection;
 using Microsoft.CSharp;
 using NUnit.Framework;
 using Avro.Specific;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
```
[the remark this answers is not in the document]
Comment (massada) [reply lang/csharp/src/apache/test/CodGen/CodeGenTest.cs]: I've rebased and this change is gone

**Comment by massada** [comment]
@KyleSchoonover rebased

**Review by KyleSchoonover — APPROVED** [review]

**Review by zcsizmadia — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @massada !

**Pull request merged**